### PR TITLE
Homework 3 is completed

### PR DESCRIPTION
**Pull request by serafimovmax**
Дмитрий, привет! Если можно, отправь пожалуйста, на проверку Семену. Спасибо!

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/driver/DriverSingelton.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    private DriverSingelton() {}
+
+    public static WebDriver getWebDriver() {
+        if (null == webDriver) {
+            WebDriverManager.chromedriver().setup();
+            webDriver = new ChromeDriver();
+            webDriver.manage().window().maximize();
+            webDriver.manage().timeouts()
+                     .implicitlyWait(10L, TimeUnit.SECONDS);
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/driver/DriverSingelton.java:20]: Move 10L to the constant

File: src/test/java/com/epam/tc/hw3/BaseHomePageTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        softAssert.assertTrue(mainPage.fullUserName().isDisplayed());
+        softAssert.assertEquals(mainPage.getUserName(), Constants.USER_FULL_NAME.get());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown() {
+        softAssert = null;
+        loginPage = null;
+        mainPage = null;
+        someUser = null;
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/BaseHomePageTest.java:49]: Do you really need these assignments?

File: src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    public List<String> getLogsComponetText() {
+        return logsComponent.stream()
+                            .map(logsText -> logsText
+                                .getText().substring(logsText.getText().indexOf(" ") + 1))
+                            .collect(Collectors.toList());
+    }
+}
+
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java:84]: Add a blank line instead this

File: src/test/java/com/epam/tc/hw3/page/LoginPage.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+
+    public MainPage login(User user) {
+        userIcon.click();
+        loginField.sendKeys(user.getLogin());
+        passwordField.sendKeys(user.getPassword());
+        loginButton.click();
+        return new MainPage(webDriver);
+    }
+
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/page/LoginPage.java:50]: remove this blank line

File: src/test/java/com/epam/tc/hw3/page/LoginPage.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        webDriver.navigate().to(URL);
+        new WebDriverWait(webDriver, 10);
+        return this;
+    }
+
+    public String getPageTitle() {
+        return webDriver.getTitle();
+    }
+
+
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/page/LoginPage.java:42]: remove this blank line

File: src/test/java/com/epam/tc/hw3/service/UserCreator.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+public class UserCreator {
+    public static final String LOGIN = Constants.LOGIN.get();
+    public static final String USER_PASSWORD = Constants.PASSWORD.get();
+
+    public static User withCredentialsFromProperty() {
+        return new User(LOGIN, USER_PASSWORD);
+    }
+
+
+
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/service/UserCreator.java:15]: remove this blank lines

File: src/test/resources/hw3/HomePageTestSuite [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="Home Page One Test Suite">
+
+  <test name="Home Page One">
+    <classes>
+      <class name="com.epam.tc.hw3.ex1.HomePageOneTest"/>
+    </classes>
+  </test>
+
+</suite>
```
Comment (komyagin) [inline src/test/resources/hw3/HomePageTestSuite:10]: Add a new blank line

File: src/test/resources/hw3/HomePageTwoTestSuite [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="Home Page Two Test Suite">
+
+  <test name="Home Page Two">
+    <classes>
+      <class name="com.epam.tc.hw3.ex2.HomePageTwoTest"/>
+    </classes>
+  </test>
+
+</suite>
```
Comment (komyagin) [inline src/test/resources/hw3/HomePageTwoTestSuite:10]: Add a new blank line

File: src/test/java/com/epam/tc/hw3/ex2/HomePageTwoTest.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+        //8. Select in dropdown
+        diffElPage.clickDropdownColors();
+        diffElPage.selectYellowColor();
+
+        //9. Assert log rows
+        softAssert.assertEquals(diffElPage.getLogsComponetText(), LOGS_TEXT);
+
+        softAssert.assertAll();
+
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/ex2/HomePageTwoTest.java:35]: remove this blank line

File: src/test/java/com/epam/tc/hw3/ex1/HomePageOneTest.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+        //10. Assert that driver has focus on the original window
+        mainPage.switchToParentFrame();
+
+        //11. Assert that left section menu items are displayed and have proper text
+        softAssert.assertTrue(mainPage.leftMenuDisplayed());
+        softAssert.assertEquals(mainPage.getLeftMenuText(), LEFT_BUTTONS_TEXT);
+
+        softAssert.assertAll();
+
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/ex1/HomePageOneTest.java:41]: remove this blank line

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw3/driver/DriverSingelton.java (hunk 1 above)
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/driver/DriverSingelton.java]: Added

File: src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java (hunk 3 above)
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java:83]: Done

File: src/test/java/com/epam/tc/hw3/page/LoginPage.java (hunk 5 above)
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/page/LoginPage.java]: Done

File: src/test/java/com/epam/tc/hw3/page/MainPage.java [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+        return leftMenu.stream()
+                       .allMatch(WebElement::isDisplayed);
+    }
+
+    public List<String> getLeftMenuText() {
+        return leftMenu.stream().map(WebElement::getText)
+                       .collect(Collectors.toList());
+    }
+
+
```
[the remark this answers is not in the document]
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/page/MainPage.java]: Done

File: src/test/java/com/epam/tc/hw3/service/UserCreator.java (hunk 6 above)
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/service/UserCreator.java]: Done

File: src/test/resources/hw3/HomePageTestSuite (hunk 7 above)
Comment (serafimovmax) [reply src/test/resources/hw3/HomePageTestSuite]: Done

File: src/test/java/com/epam/tc/hw3/ex2/HomePageTwoTest.java (hunk 9 above)
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/ex2/HomePageTwoTest.java]: Done

File: src/test/java/com/epam/tc/hw3/ex1/HomePageOneTest.java (hunk 10 above)
Comment (serafimovmax) [reply src/test/java/com/epam/tc/hw3/ex1/HomePageOneTest.java]: Done

**Review by komyagin — APPROVED** [review]

**Pull request merged**